src/Rings/mpoly.jl

```diff
@@ -512,11 +512,12 @@ singular_poly_ring(R::Singular.PolyRing; keep_ordering::Bool = true) = R
 function singular_coeff_ring(K::AbsSimpleNumField)
   minpoly = defining_polynomial(K)
   Qa = parent(minpoly)
-  a = gen(Qa)
   SQa, (Sa,) = Singular.FunctionField(Singular.QQ, _variables_for_singular(symbols(Qa)))
   Sminpoly = SQa(coeff(minpoly, 0))
+  var = Sa
   for i in 1:degree(minpoly)
-    Sminpoly += SQa(coeff(minpoly, i))*Sa^i
+      Sminpoly = addmul!(Sminpoly, SQa(coeff(minpoly, i)), var)
+      var *= Sa
```

Comment:
```suggestion
      var = mul!(var, var, Sa)
```
might also speed things up

Reply:
or just
```suggestion
      var = mul!(var, Sa)
```
and then the same everywhere else

Reply:
After testing, this is indeed faster and I have included this